utils/src/pickTreeItem/contextValue/compatibility/CompatibilityRecursiveQuickPickStep.ts

```diff
@@ -29,6 +29,25 @@ interface CompatibilityRecursiveQuickPickOptions extends types.ContextValueFilte
 export class CompatibilityRecursiveQuickPickStep<TContext extends types.QuickPickWizardContext & types.ITreeItemPickerContext> extends CompatibilityContextValueQuickPickStep<TContext, CompatibilityRecursiveQuickPickOptions> {
 
     protected override async promptInternal(wizardContext: TContext): Promise<unknown> {
+
+        const lastPickedItem = getLastNode(wizardContext);
+        const lastPickedItemTi = isWrapper(lastPickedItem) ? lastPickedItem.unwrap<AzExtTreeItem>() : lastPickedItem;
+
+        this.promptOptions = isAzExtParentTreeItem(lastPickedItemTi) ? {
```

Comment:
Can't we just edit the properties rather than recreating the object? Then `promptOptions` can stay `readonly` in `GenericQuickPickStep`.

Reply:
Fixed. That's why we do code reviews! 😄